src/java/org/apache/cassandra/io/sstable/SSTableZeroCopyWriter.java

```diff
@@ -233,4 +237,22 @@ private void write(AsyncStreamingInputPlus in, long size, SequentialWriter write
             throw new FSWriteError(e, writer.getPath());
         }
     }
+
+    private static class ZeroCopySequentialWriter extends SequentialWriter
+    {
+        private ZeroCopySequentialWriter(File file, SequentialWriterOption option, boolean strictFlushing)
+        {
+            super(file, ByteBufferUtil.EMPTY_BYTE_BUFFER, option, strictFlushing);
+        }
+
+//        @Override
+//        public void write(byte[] b, int off, int len) throws IOException
+//        {
+//            // This logic is never valid in production environments and is only here for unit tests.  By default we never
+//            // allocate this buffer, but in unit tests we need one, so allocate to make those unit tests happy.
```

Comment:
Maybe rephrase and move this comment up to the method. In production, we do not expect this entire method to be called, as only `writeDirectlyToChannel` should be invoked for zero-copy